src/main/java/uk/gov/hmcts/reform/civil/exceptions/CaseIdNotProvidedException.java

```diff
@@ -0,0 +1,18 @@
+package uk.gov.hmcts.reform.civil.exceptions;
+
+public class CaseIdNotProvidedException extends NotRetryableException {
+
+    private static final String ERROR_MESSAGE = "The caseId was not provided";
+
+    public CaseIdNotProvidedException() {
+        super(ERROR_MESSAGE);
+    }
+
+    public CaseIdNotProvidedException(String message) {
+        super(message);
+    }
+
+    public CaseIdNotProvidedException(String message, Throwable cause) {
```

Comment:
I don't think there is a need of CaseIdNotProvidedException as this will always be there, once a case is initiated this is the generated straight away in the system

Reply:
This is provided as SonarLint complains of some "possible" null values. This situation will probably never happen, but if it does there's definitely something wrong with the data.

We can probably remove the dedicated exception without sacrificing functionality. I'll take a look.